plugins/module_utils/_vmware_facts.py

```diff
@@ -281,6 +281,128 @@ def hw_network_device_facts(self):
         }
 
 
+class ClusterFacts():
+    def __init__(self, cluster):
+        self.cluster = cluster
+
+    def all_facts(self):
+        return {
+            **self.host_facts(),
+            **self.ha_facts(),
+            **self.identifier_facts(),
+            **self.drs_facts(),
+            **self.vsan_facts(),
+            **self.resource_usage_facts(),
+            **self.dpm_facts()
+        }
+
+    def identifier_facts(self):
+        return {
+            "moid": self.cluster._moId,
+            "datacenter": self.cluster.parent.parent.name
+        }
+
+    def host_facts(self):
+        hosts = []
+        for host in self.cluster.host:
+            hosts.append({
+                'name': host.name,
+                'folder': get_folder_path_of_vm(host),
```

Comment:
The function is called get_folder_path_of_vm but it gets an host.... why?

Reply:
I think when it was copied it just looked up vm paths but now its more generic.
I updated the name